pkg/scanner/utils/utils_test.go

```diff
@@ -0,0 +1,80 @@
+package utils
+
+import (
+	"testing"
+
+	"github.com/knqyf263/go-version"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMatchVersions2(t *testing.T) {
+	testCases := []struct {
+		name           string
+		currentVersion string
+		rangeVersion   []string
+		expectedCheck  bool
+	}{
+		{
+			name:           "pass: expect true when os/machine is in pre-release",
+			currentVersion: "1.9.25-x86-mingw32",
+			rangeVersion:   []string{`>= 1.9.24`},
+			expectedCheck:  true,
+		},
+		{
+			name:           "pass: expect true when language is in pre-release",
+			currentVersion: "1.8.6-java",
+			rangeVersion:   []string{`~> 1.5.5`, `~> 1.6.8`, `>= 1.7.7`},
+			expectedCheck:  true,
+		},
+		{
+			name:           "expect false",
+			currentVersion: "1.9.23-x86-mingw32",
+			rangeVersion:   []string{`>= 1.9.24`},
+			expectedCheck:  false,
+		},
+		{
+			// passes if (>= 1.2.3, < 2.0.0)
+			name:           "expect false",
+			currentVersion: "1.2.4",
+			rangeVersion:   []string{`^1.2.3`},
+			expectedCheck:  true,
+		},
+		{
+			// passes if (>= 1.2.3, < 2.0.0)
+			name:           "expect false",
+			currentVersion: "2.0.0",
+			rangeVersion:   []string{`^1.2.3`},
+			expectedCheck:  false,
+		},
+		{
+			// passes if (>= 2.0.18, < 3.0.0) || (>= 3.1.16, < 4.0.0) || (>= 4.0.8, < 5.0.0) || ( >=5.0.0,<6.0.0)
+			name:           "expect false",
+			currentVersion: "3.1.16",
+			rangeVersion:   []string{`^2.0.18 || ^3.1.6 || ^4.0.8 || ^5.0.0-beta.5`},
+			expectedCheck:  true,
+		},
+		{
+			// passes if (>= 2.0.18, < 3.0.0) || (>= 3.1.16, < 4.0.0) || (>= 4.0.8, < 5.0.0) || ( >=5.0.0,<6.0.0)
+			name:           "expect false",
+			currentVersion: "6.0.0",
+			rangeVersion:   []string{`^2.0.18 || ^3.1.6 || ^4.0.8 || ^5.0.0-beta.5`},
+			expectedCheck:  false,
+		},
+		{
+			// passes if (>= 2.0.18, < 3.0.0) || (>= 3.1.16, < 4.0.0) || (>= 4.0.8, < 5.0.0) || ( >=5.0.0,<6.0.0)
+			name:           "expect false",
+			currentVersion: "5.0.0-beta.5",
+			rangeVersion:   []string{`^2.0.18 || ^3.1.6 || ^4.0.8 || ^5.0.0-beta.5`},
+			expectedCheck:  true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			v, err := version.NewVersion(tc.currentVersion)
+			assert.Nil(t, err)
```

Comment:
`assert` goes through even if it doesn't satisfy the condition.

```suggestion
			require.NoError(t, err)
```